pkg/machinehealthcheck/machinehealthcheck.go

```diff
@@ -56,21 +56,23 @@ var _ = Describe("MachineHealthCheck", framework.LabelMachineHealthChecks, func(
 		Expect(err).ToNot(HaveOccurred(), "failed to create a new machineSet resource")
 
 		framework.WaitForMachineSet(ctx, client, machineSet.GetName())
+
+		DeferCleanup(func() {
+			By("Deleting the MachineHealthCheck resource")
+			Expect(client.Delete(context.Background(), machinehealthcheck)).To(Succeed(), "failed to delete MHC")
+
+			By("Deleting the new MachineSet")
+			Expect(client.Delete(context.Background(), machineSet)).To(Succeed(), "failed to delete machineSet")
+
+			framework.WaitForMachineSetsDeleted(ctx, client, machineSet)
+		})
```

Comment:
This should come before the `WaitForMachineSet` else it won't run if the wait fails